src/gui/tray/Window.qml

```diff
@@ -459,7 +459,17 @@ Window {
                         Menu {
                             id: appsMenu
                             y: (trayWindowAppsButton.y + trayWindowAppsButton.height + 2)
-                            width: (Style.headerButtonIconSize * 3)
+                            width: {

```

Comment:
Although it kinds of upset me that we need to specify that width at all... but I guess it's just QQC2 being somewhat odd. I think we'd have better results using:
`width: Math.min(contentItem.childrenRect.width, Style.trayWindowWidth / 2)`

It'd give a different look to what you got (larger entries by default) but at least that respect the styling of the menu items. If we want those menu items to be smaller like you did, we could always have something like this in the MenuItem based delegate:
`width: contentItem.implicitWidth + leftPadding + rightPadding`

Should be faster too avoiding a loop and several JS calls.

Reply:
This is awesome! Thanks!

Reply:
Btw, without changing the `MenuItem`s' `width`, the `childrenRect.width` was already half the tray window width for our standard app list. This looked ugly, so I made it look again as in the screenshots above by changing the `MenuItem` width as you suggested.

Reply:
Yep, it was half the tray window width indeed, but by chance! ;-)

Just so happen that trayWindowWidth is 400 and that the default style in QQC2 uses 200 as implicitWidth for the default MenuItem background (so it'll tend to be its minimum width).